opentelemetry-sdk/src/metrics/mod.rs

```diff
@@ -1046,39 +1053,144 @@ mod tests {
+            find_histogram_datapoint_with_key_value(&histogram_data.data_points, "key1", "value2")
                 .expect("datapoint with key1=value2 expected");
         assert_eq!(data_point2.count, values_kv2.len() as u64);
         assert_eq!(data_point2.sum, values_kv2.iter().sum::<u64>());
         assert_eq!(data_point2.min.unwrap(), *values_kv2.iter().min().unwrap());
         assert_eq!(data_point2.max.unwrap(), *values_kv2.iter().max().unwrap());
+
+        // Reset and report more measurements
+        test_context.reset_metrics();
+        for value in values_kv1.iter() {
+            histogram.record(*value, &[KeyValue::new("key1", "value1")]);
+        }
+
+        for value in values_kv2.iter() {
+            histogram.record(*value, &[KeyValue::new("key1", "value2")]);
+        }
+
+        test_context.flush_metrics();
+
+        let histogram_data =
+            test_context.get_aggregation::<data::Histogram<u64>>("my_histogram", None);
+        assert_eq!(histogram_data.data_points.len(), 2);
+        let data_point1 =
+            find_histogram_datapoint_with_key_value(&histogram_data.data_points, "key1", "value1")
+                .expect("datapoint with key1=value1 expected");
+        if temporality == Temporality::Cumulative {
+            assert_eq!(data_point1.count, 2 * (values_kv1.len() as u64));
+            assert_eq!(data_point1.sum, 2 * (values_kv1.iter().sum::<u64>()));
+            assert_eq!(data_point1.min.unwrap(), *values_kv1.iter().min().unwrap());
+            assert_eq!(data_point1.max.unwrap(), *values_kv1.iter().max().unwrap());
+        } else {
+            assert_eq!(data_point1.count, values_kv1.len() as u64);
+            assert_eq!(data_point1.sum, values_kv1.iter().sum::<u64>());
+            assert_eq!(data_point1.min.unwrap(), *values_kv1.iter().min().unwrap());
+            assert_eq!(data_point1.max.unwrap(), *values_kv1.iter().max().unwrap());
+        }
+
+        let data_point1 =
+            find_histogram_datapoint_with_key_value(&histogram_data.data_points, "key1", "value2")
+                .expect("datapoint with key1=value1 expected");
+        if temporality == Temporality::Cumulative {
+            assert_eq!(data_point1.count, 2 * (values_kv2.len() as u64));
+            assert_eq!(data_point1.sum, 2 * (values_kv2.iter().sum::<u64>()));
+            assert_eq!(data_point1.min.unwrap(), *values_kv2.iter().min().unwrap());
+            assert_eq!(data_point1.max.unwrap(), *values_kv2.iter().max().unwrap());
+        } else {
+            assert_eq!(data_point1.count, values_kv2.len() as u64);
+            assert_eq!(data_point1.sum, values_kv2.iter().sum::<u64>());
+            assert_eq!(data_point1.min.unwrap(), *values_kv2.iter().min().unwrap());
+            assert_eq!(data_point1.max.unwrap(), *values_kv2.iter().max().unwrap());
+        }
+    }
+
+    fn gauge_aggregation_helper() {
+        // Arrange
+        let mut test_context = TestContext::new(Temporality::Delta);
```

Comment:
We should test this for both `Delta` and `Cumulative` to ensure that Gauge always uses last value aggregation regardless of the aggregation temporality of the `MetricReader`.

Reply:
+1